cmake/tests/stable_inplace_merge.cpp

```diff
@@ -0,0 +1,15 @@
+////////////////////////////////////////////////////////////////////////////////
+//  Copyright (c) 2017 Mikael Simberg
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+////////////////////////////////////////////////////////////////////////////////
+
+#if defined(_LIBCPP_VERSION) && (_LIBCPP_VERSION < 6)
+#  error "libc++ inplace_merge implementation is broken"
+#endif
```

Comment:
Instead of checking the stdlib version, couldn't we actually test if inplace_merge is actually stable?

Reply:
Right, that does sound like the correct thing to do. Need to make sure it's thorough enough to catch future bugs in `inplace_merge` as well though, so to be honest I think the current check is simple and good enough. (libc++ is unlikely to have this bug again if they've added good tests for it, but then we need not trust them of course...)